socorro/external/es/correlations.py

```diff
@@ -51,6 +84,124 @@ def create_correlations_index(self, es_index):
             es_settings = json.loads(settings_json)
 
             index_creator = self.config.index_creator(self.config)
+            # This is resilient to being called repeatedly even
+            # when the index already exists.
             index_creator.create_index(es_index, es_settings)
-
             self.indices_cache.add(es_index)
+
+    def _prefix_to_datetime_date(self, prefix):
+        yy = int(prefix[:4])
+        mm = int(prefix[4:6])
+        dd = int(prefix[6:8])
+        return datetime.date(yy, mm, dd)
+
+    @staticmethod
+    def make_id(document):
+        string_parts = []
+        for key, value in sorted(document.items()):
+            if isinstance(value, datetime.date):
+                value = value.isoformat()
+            if isinstance(value, int):
+                value = str(value)
+            if isinstance(value, basestring):
+                string_parts.append(value)
+        return hashlib.md5(
+            (''.join(string_parts)).encode('utf-8')
+        ).hexdigest()
+
+
+class CoreCounts(Correlations):
+
+    def store(
+        self,
+        counts_summary_structure,
+        prefix,
+        name,
+        key,
+    ):
+        date = self._prefix_to_datetime_date(prefix)
+        index = self.get_index_for_date(date)
+        self.create_correlations_index(index)
+
+        notes = counts_summary_structure['notes']
+        product = key.split('_')[0]
+        version = key.split('_')[1]
+        for platform in counts_summary_structure:
+            if platform not in self.config.recognized_platforms:
+                continue
+            count = counts_summary_structure[platform]['count']
+            signatures = counts_summary_structure[platform]['signatures']
+
+            for signature, payload in signatures.items():
+                doc = {
+                    'platform': platform,
+                    'product': product,
+                    'version': version,
+                    'count': count,
+                    'signature': signature,
+                    'payload': json.dumps(payload),
+                    'date': date,
+                    'key': name,
+                    'notes': notes,
+                }
+                with self.es_context() as conn:
+                    conn.index(
+                        index=index,
+                        # see correlations_index_settings.json
+                        doc_type='correlations',
+                        body=doc,
+                        id=self.make_id(doc)
```

Comment:
Maybe what you want to solve is the problem of the job failing midway. If that happens, and you re-run it from the start, and you use random IDs, then the same document might be indexed twice. However, if that's what you're chasing, I think there is another problem: what happens if you re-run the correlations and the results are different? Then all of your IDs will be different, and you will have a lot of duplicated results. 

I think that, in that case, you want to generate the ID based on keys that actually make a good primary key. That would be ``product``, ``version``, ``platform``, ``date`` and ``signature``. ``count``, ``notes`` and ``payload``, notably, shouldn't be part of the ID since they could potentially change.